README.md
# [li_info](https://pub.dev/packages/li_info) Getting Started

This project is a starting point for a Flutter
[plug-in package](https://flutter.dev/developing-packages/),
a specialized package that includes platform-specific implementation code for
Android and/or iOS.

For help getting started with Flutter, view our
[online documentation](https://flutter.dev/docs), which offers tutorials,
samples, guidance on mobile development, and a full API reference.


## Plugin android and ios
<br><img src="documentation_images/android_li_plugin_0.0.1.gif" width="250" height="500">
&nbsp;&nbsp;&nbsp;<img src="documentation_images/ios_li_plugin_0.0.9.gif" width="250" height="500">

also most welcome for contributors [Contact Us](mailto:kishor@logisticinfotech.com) 



CHANGELOG.md
## [0.0.9] - add image selection permission in ios   
## [0.0.8] - add custom textField widgets and add image selection with cropping  
## [0.0.7] - add custom textField widgets and add image selection with cropping  
## [0.0.6] - add more buttons
## [0.0.5] - remove third party plugin and make buttons
## [0.0.4] - add social media button
## [0.0.3] - add null-safety
## [0.0.2] - Update plugin and add get version code
## [0.0.1] - add for first publish testing after we have work on that  


example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:li_info_example/main.dart';

void main() {
  testWidgets('Verify Platform version', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(MyApp());

    // Verify that platform version is retrieved.
    expect(
      find.byWidgetPredicate(
        (Widget widget) => widget is Text &&
                           widget.data!.startsWith('Running on:'),
      ),
      findsOneWidget,
    );
  });
}


example/lib/main.dart
import 'dart:io';

import 'package:flutter/material.dart';
import 'package:li_info/li_info.dart';

// main method run app form here
void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return PluginExample();
  }
}

class PluginExample extends StatefulWidget {
  @override
  _PluginExampleState createState() => _PluginExampleState();
}

class _PluginExampleState extends State<PluginExample> {
  String label = "Clicked button!";
  String? imagePath = null;
  var emailEditingController = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: true,
      title: 'Flutter Demo',
      theme: ThemeData(
        iconTheme: IconThemeData(
          color: Colors.white,
        ),
        accentColor: Colors.red,
      ),
      home: Scaffold(
        body: Container(
          child: Center(
            child: SingleChildScrollView(
              child: Column(
                mainAxisAlignment: MainAxisAlignment.center,
                children: <Widget>[
                  Text(label),
                  SizedBox(height: 10.0),
                  EditText(
                    margin: EdgeInsets.all(10.0),
                    focusColor: Color(0xFFEC3636),
                    style: TextStyle(color: Color(0xFF000000)),
                    hintText: "Enter email",
                    textEditingController: emailEditingController,
                    unFocusColor: Color(0xFFFFFF),
                    textInputAction: TextInputAction.done,
                    keyboardType: TextInputType.emailAddress,
                  ),
                  SizedBox(height: 10.0),
                  SignInButton(
                    Buttons.KishorParmar,
                    text: imagePath != null ? "Remove File" : "Select Image",
                    onPressed: () async {
                      if (imagePath != null) {
                        setState(() {
                          imagePath = null;
                        });
                      } else {
                        var temp = await getFile();
                        setState(() {
                          imagePath = (temp != null ? temp : null)!;
                          label = 'File Path: ${temp != null ? temp : ""}';
                        });
                      }
                    },
                  ),
                  imagePath != null ? Image.file(File(imagePath!)) : Container(),
                  SignInButton(
                    Buttons.Google,
                    onPressed: () {
                      setState(() {
                        label = 'Google';
                      });
                    },
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.all(Radius.circular(20)),
                    ),
                  ),
                  SizedBox(
                    height: 10,
                  ),
                  SignInButton(
                    Buttons.Facebook,
                    onPressed: () {
                      setState(() {
                        label = 'Facebook';
                      });
                    },
                  ),
                  SizedBox(
                    height: 10,
                  ),
                  SignInButton(
                    Buttons.Apple,
                    onPressed: () {
                      setState(() {
                        label = 'Apple';
                      });
                    },
                  ),
                  SizedBox(
                    height: 10,
                  ),
                  SignInButton(
                    Buttons.Microsoft,
                    onPressed: () {
                      setState(() {
                        label = 'Microsoft';
                      });
                    },
                  ),
                  SizedBox(
                    height: 10,
                  ),
                  SignInButton(
                    Buttons.GitHub,
                    text: "Sign up with GitHub",
                    onPressed: () {
                      setState(() {
                        label = 'Github';
                      });
                    },
                  ),
                  SizedBox(
                    height: 10,
                  ),
                  SignInButton(
                    Buttons.Twitter,
                    text: "Sign In with Twitter",
                    onPressed: () {
                      setState(() {
                        label = 'Twitter';
                      });
                    },
                  ),
                  SignInButton(
                    Buttons.KishorParmar,
                    text: "Custom button",
                    onPressed: () {
                      setState(() {
                        label = 'Custom button';
                      });
                    },
                  ),
                  SizedBox(
                    height: 10,
                  ),
                ],
              ),
            ),
          ),
        ),
      ),
    );
  }
}


lib/social_signin_buttons.dart
library li_info;

import 'package:flutter/material.dart';
import 'package:font_awesome_flutter/font_awesome_flutter.dart';

enum Buttons {
  Google,
  Facebook,
  Microsoft,
  LinkedIn,
  Twitter,
  GitHub,
  Apple,
  Email,
}

class SignInButton extends StatelessWidget {
  final Function? onPressed;
  final Buttons? button;
  final String? text;
  final ShapeBorder? shape;

  SignInButton(this.button, {@required this.onPressed, this.text, this.shape})
      : assert(button != null),
        assert(onPressed != null);

  @override
  Widget build(BuildContext context) {
    switch (button) {
      case Buttons.Email:
        return Btn(
          key: ValueKey("Email"),
          text: text ?? "Sign In With Email",
          icon: Icons.email,
          onPressed: onPressed,
          backgroundColor: Colors.white,
          textColor: Colors.black,
          iconColor: Colors.black,
          shape: shape,
        );
      case Buttons.Google:
        return Btn(
          key: ValueKey("Google"),
          text: text ?? "Sign In With Google",
          image: Container(
            margin: EdgeInsets.fromLTRB(0.0, 0.0, 5.0, 0.0),
            child: ClipRRect(
              borderRadius: BorderRadius.circular(8.0),
              child: Image(
                image: AssetImage(
                  'assets/google_logo.png',
                  package: 'li_info',
                ),
                height: 25.0,
              ),
            ),
          ),
          onPressed: onPressed,
          backgroundColor: Colors.white,
          textColor: Colors.black,
          iconColor: Colors.black,
          shape: shape,
        );
      case Buttons.Facebook:
        return Btn(
          key: ValueKey("Facebbok"),
          text: text ?? "Sign In With Facebook",
          icon: FontAwesomeIcons.facebookF,
          onPressed: onPressed,
          backgroundColor: Color(0xFF3B5998),
          textColor: Colors.white,
          iconColor: Colors.white,
          shape: shape,
        );
      case Buttons.GitHub:
        return Btn(
          key: ValueKey("Github"),
          text: text ?? "Sign In With Github",
          icon: FontAwesomeIcons.github,
          onPressed: onPressed,
          backgroundColor: Color(0xFF444444),
          textColor: Colors.white,
          iconColor: Colors.white,
          shape: shape,
        );
      case Buttons.Apple:
        return Btn(
          key: ValueKey("Apple"),
          text: text ?? "Sign In With Apple",
          icon: FontAwesomeIcons.apple,
          onPressed: onPressed,
          backgroundColor: Colors.white,
          textColor: Colors.black,
          iconColor: Colors.black,
          shape: shape,
        );
      case Buttons.LinkedIn:
        return Btn(
          key: ValueKey("LinkedIn"),
          text: text ?? "Sign In With LinkedIn",
          icon: FontAwesomeIcons.linkedin,
          onPressed: onPressed,
          backgroundColor: Color(0xFF007BB6),
          textColor: Colors.white,
          iconColor: Colors.white,
          shape: shape,
        );
      case Buttons.Microsoft:
        return Btn(
          key: ValueKey("Microsoft"),
          text: text ?? "Sign In With Microsoft",
          image: Container(
            margin: EdgeInsets.fromLTRB(0.0, 0.0, 5.0, 0.0),
            child: ClipRRect(
              borderRadius: BorderRadius.circular(0.0),
              child: Image(
                image: AssetImage(
                  'assets/microsoft_logo.png',
                  package: 'li_info',
                ),
                height: 27.0,
              ),
            ),
          ),
          onPressed: onPressed,
          backgroundColor: Colors.white,
          textColor: Colors.black,
          iconColor: Colors.black,
          shape: shape,
        );
      case Buttons.Twitter:
        return Btn(
          key: ValueKey("Twitter"),
          text: text ?? "Sign In With Twitter",
          icon: FontAwesomeIcons.twitter,
          onPressed: onPressed,
          backgroundColor: Color(0xFF1DA1F2),
          textColor: Colors.white,
          iconColor: Colors.white,
          shape: shape,
        );
      default: {
        return Container();
      }
    }
  }
}

class Btn extends StatelessWidget {
  final Color? textColor, backgroundColor, iconColor;
  final Function? onPressed;
  final String? text;
  final IconData? icon;
  final Widget? image;
  final ShapeBorder? shape;

  Btn({Key? key, this.backgroundColor, this.onPressed, this.text, this.icon, this.textColor, this.iconColor, this.image, this.shape});

  @override
  Widget build(BuildContext context) {
    return MaterialButton(
      key: key,
      onPressed: () => onPressed,
      color: backgroundColor,
      shape: shape ?? ButtonTheme.of(context).shape,
      padding: const EdgeInsets.all(0),
      child: Container(
        constraints: BoxConstraints(
          maxWidth: 220,
        ),
        child: Center(
          child: Row(
            crossAxisAlignment: CrossAxisAlignment.center,
            children: <Widget>[
              Padding(
                padding: EdgeInsets.symmetric(
                  horizontal: 13,
                ),
                child: image != null
                    ? image
                    : Icon(
                        icon,
                        size: 20,
                        color: this.iconColor,
                      ),
              ),
              Text(
                text!,
                style: TextStyle(
                  fontFamily: 'Roboto',
                  color: textColor,
                  fontSize: 14,
                  backgroundColor: Color.fromRGBO(0, 0, 0, 0),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}


android/src/main/kotlin/com/example/li_info/LiInfoPlugin.kt
package com.example.li_info

import androidx.annotation.NonNull

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** LiInfoPlugin */
class LiInfoPlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "li_info")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
    if (call.method == "getPlatformVersion") {
      result.success("Android ${android.os.Build.VERSION.RELEASE}")
    } else {
      result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}


lib/li_info.dart
library li_info;

import 'dart:io';

import 'package:dio/dio.dart';
import 'package:flutter/material.dart';
import 'package:images_picker/images_picker.dart';

enum Buttons {
  Google,
  Facebook,
  Microsoft,
  LinkedIn,
  Twitter,
  GitHub,
  Apple,
  Email,
  KishorParmar,
}

/*
*
* Custom Button widgets
*
* */

class SignInButton extends StatelessWidget {
  final Function? onPressed;
  final Buttons? button;
  final String? text;
  final ShapeBorder? shape;
  final IconData? icon;

  SignInButton(this.button, {@required this.onPressed, this.text, this.shape, this.icon})
      : assert(button != null),
        assert(onPressed != null);

  @override
  Widget build(BuildContext context) {
    switch (button) {
      case Buttons.Email:
        return Btn(
          key: ValueKey("Email"),
          text: text ?? "Sign In With Email",
          icon: Icons.email,
          onPressed: () {
            if (onPressed != null) {
              onPressed?.call();
            }
          },
          backgroundColor: Colors.white,
          textColor: Colors.black,
          iconColor: Colors.black,
          shape: shape,
        );
      case Buttons.Google:
        return Btn(
          key: ValueKey("Google"),
          text: text ?? "Sign In With Google",
          image: Container(
            margin: EdgeInsets.fromLTRB(0.0, 0.0, 5.0, 0.0),
            child: ClipRRect(
              borderRadius: BorderRadius.circular(8.0),
              child: Image(
                image: AssetImage(
                  'assets/google_logo.png',
                  package: 'li_info',
                ),
                height: 25.0,
              ),
            ),
          ),
          onPressed: () {
            if (onPressed != null) {
              onPressed?.call();
            }
          },
          backgroundColor: Colors.white,
          textColor: Colors.black,
          iconColor: Colors.black,
          shape: shape,
        );
      case Buttons.Facebook:
        return Btn(
          key: ValueKey("Facebook"),
          text: text ?? "Sign In With Facebook",
          icon: icon,
          onPressed: () {
            if (onPressed != null) {
              onPressed?.call();
            }
          },
          backgroundColor: Color(0xFF3B5998),
          textColor: Colors.white,
          iconColor: Colors.white,
          shape: shape,
        );
      case Buttons.GitHub:
        return Btn(
          key: ValueKey("Github"),
          text: text ?? "Sign In With Github",
          icon: icon,
          onPressed: () {
            if (onPressed != null) {
              onPressed?.call();
            }
          },
          backgroundColor: Color(0xFF444444),
          textColor: Colors.white,
          iconColor: Colors.white,
          shape: shape,
        );
      case Buttons.Apple:
        return Btn(
          key: ValueKey("Apple"),
          text: text ?? "Sign In With Apple",
          icon: icon,
          onPressed: () {
            if (onPressed != null) {
              onPressed?.call();
            }
          },
          backgroundColor: Colors.white,
          textColor: Colors.black,
          iconColor: Colors.black,
          shape: shape,
        );
      case Buttons.LinkedIn:
        return Btn(
          key: ValueKey("LinkedIn"),
          text: text ?? "Sign In With LinkedIn",
          icon: icon,
          onPressed: () {
            if (onPressed != null) {
              onPressed?.call();
            }
          },
          backgroundColor: Color(0xFF007BB6),
          textColor: Colors.white,
          iconColor: Colors.white,
          shape: shape,
        );
      case Buttons.Microsoft:
        return Btn(
          key: ValueKey("Microsoft"),
          text: text ?? "Sign In With Microsoft",
          image: Container(
            margin: EdgeInsets.fromLTRB(0.0, 0.0, 5.0, 0.0),
            child: ClipRRect(
              borderRadius: BorderRadius.circular(0.0),
              child: Image(
                image: AssetImage(
                  'assets/microsoft_logo.png',
                  package: 'li_info',
                ),
                height: 27.0,
              ),
            ),
          ),
          onPressed: () {
            if (onPressed != null) {
              onPressed?.call();
            }
          },
          backgroundColor: Colors.white,
          textColor: Colors.black,
          iconColor: Colors.black,
          shape: shape,
        );
      case Buttons.Twitter:
        return Btn(
          key: ValueKey("Twitter"),
          text: text ?? "Sign In With Twitter",
          icon: icon,
          onPressed: () {
            if (onPressed != null) {
              onPressed?.call();
            }
          },
          backgroundColor: Color(0xFF1DA1F2),
          textColor: Colors.white,
          iconColor: Colors.white,
          shape: shape,
        );
      case Buttons.KishorParmar:
        return Btn(
          key: ValueKey("KishorParmar"),
          text: text ?? "Kishor Parmar",
          icon: icon,
          onPressed: () {
            if (onPressed != null) {
              onPressed?.call();
            }
          },
          backgroundColor: Color(0xFFFF0A0A),
          textColor: Colors.white,
          iconColor: Colors.white,
          shape: shape,
        );
      default:
        {
          return Container();
        }
    }
  }
}

class Btn extends StatelessWidget {
  final Color? textColor, backgroundColor, iconColor;
  final Function? onPressed;
  final String? text;
  final IconData? icon;
  final Widget? image;
  final ShapeBorder? shape;

  Btn({Key? key, this.backgroundColor, this.onPressed, this.text, this.icon, this.textColor, this.iconColor, this.image, this.shape});

  @override
  Widget build(BuildContext context) {
    return MaterialButton(
      key: key,
      onPressed: () {
        if (onPressed != null) {
          onPressed?.call();
        }
      },
      color: backgroundColor,
      shape: shape ?? ButtonTheme.of(context).shape,
      padding: const EdgeInsets.all(0),
      child: Container(
        constraints: BoxConstraints(
          maxWidth: 220,
        ),
        child: Center(
          child: Row(
            crossAxisAlignment: CrossAxisAlignment.center,
            children: <Widget>[
              Padding(
                padding: EdgeInsets.symmetric(
                  horizontal: 13,
                ),
                child: image != null
                    ? image
                    : Icon(
                        icon,
                        size: 20,
                        color: this.iconColor,
                      ),
              ),
              Text(
                text!,
                style: TextStyle(
                  fontFamily: 'Roboto',
                  color: textColor,
                  fontSize: 14,
                  backgroundColor: Color.fromRGBO(0, 0, 0, 0),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

/*
*
* Custom TextField widgets
*
* */
//ignore: must_be_immutable
class EditText extends StatefulWidget {
  EditText({Color? focusColor, Color? unFocusColor, TextStyle? style, TextEditingController? textEditingController, String? hintText, EdgeInsets? margin, TextInputAction? textInputAction, TextInputType? keyboardType}) {
    if (focusColor != null) this.focusColor = focusColor;
    if (unFocusColor != null) this.unFocusColor = unFocusColor;
    if (style != null) this.style = style;
    if (hintText != null) this.hintText = hintText;
    if (textEditingController != null) this.textEditingController = textEditingController;
    if (margin != null) this.margin = margin;
    if (textInputAction != null) this.textInputAction = textInputAction;
    if (keyboardType != null) this.keyboardType = keyboardType;
  }

  bool isEmailFocus = false;
  Color? focusColor;
  Color? unFocusColor;
  TextStyle? style;
  String? hintText;
  var textEditingController = new TextEditingController();
  EdgeInsets? margin;
  TextInputAction? textInputAction;
  TextInputType? keyboardType;

  @override
  _EditTextState createState() => _EditTextState();
}

class _EditTextState extends State<EditText> {
  @override
  Widget build(BuildContext context) {
    return Container(
      child: Focus(
        onFocusChange: (values) {
          setState(() {
            widget.isEmailFocus = values;
          });
        },
        child: Container(
          margin: widget.margin ?? EdgeInsets.all(0.0),
          decoration: BoxDecoration(
            gradient: LinearGradient(
              colors: [
                widget.isEmailFocus ? Colors.white : Colors.transparent,
                widget.isEmailFocus ? Colors.white : Colors.transparent,
              ],
              begin: Alignment.centerLeft,
              end: Alignment.centerRight,
            ),
            borderRadius: const BorderRadius.all(
              Radius.circular(10.0),
            ),
            boxShadow: [
              BoxShadow(
                color: widget.isEmailFocus
                    ? widget.focusColor == null
                        ? Color(0xFFFFFF).withOpacity(0.1)
                        : widget.focusColor!.withOpacity(0.1)
                    : widget.unFocusColor == null
                        ? Color(0xFFFFFF).withOpacity(0)
                        : widget.unFocusColor!.withOpacity(0),
                spreadRadius: 4,
                blurRadius: 10,
                offset: Offset(0, 3),
              )
            ],
          ),
          child: Container(
            color: Colors.transparent,
            child: TextField(
              onSubmitted: (_) => FocusScope.of(context).nextFocus(),
              controller: widget.textEditingController,
              keyboardType: widget.keyboardType,
              textInputAction: widget.textInputAction,
              style: widget.style,
              autocorrect: true,
              decoration: InputDecoration(
                contentPadding: EdgeInsets.symmetric(horizontal: 15.0, vertical: 25.0),
                hintText: widget.hintText,
                hintStyle: TextStyle(color: Colors.grey),
                fillColor: Colors.white70,
                enabledBorder: OutlineInputBorder(
                  borderRadius: BorderRadius.all(Radius.circular(12.0)),
                  borderSide: BorderSide(color: Color(0xFFEFEFEF), width: 2),
                ),
                focusedBorder: OutlineInputBorder(
                  borderRadius: BorderRadius.all(Radius.circular(10.0)),
                  borderSide: BorderSide(color: Color(0xFF622CFD), width: 2),
                ),
              ),
            ),
          ),
        ),
      ),
    );
  }
}

/*
*
*  this method is used to get image path from gallery
*
* */

Future<String?> getFile() async {
  String? path;
  List<Media>? res = await ImagesPicker.pick(
    count: 3,
    pickType: PickType.all,
    language: Language.System,
    // maxSize: 500,
    cropOpt: CropOption(
      aspectRatio: CropAspectRatio.wh16x9,
    ),
  );
  if (res != null) {
    print(res.map((e) => e.path).toList());

    path = res[0].thumbPath;
  }
  return path;
}

class SelectFileWidgets extends StatefulWidget {
  @override
  _SelectFileWidgets createState() => _SelectFileWidgets();
}

/*
*
*  Image selection and file downloading
*
* */
class _SelectFileWidgets extends State<SelectFileWidgets> {
  String? path;

  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Plugin example app'),
        ),
        body: Column(
          children: [
            ElevatedButton(
              child: Text('pick'),
              onPressed: () async {
                List<Media>? res = await ImagesPicker.pick(
                  count: 3,
                  pickType: PickType.all,
                  language: Language.System,
                  // maxSize: 500,
                  cropOpt: CropOption(
                    aspectRatio: CropAspectRatio.wh16x9,
                  ),
                );
                if (res != null) {
                  print(res.map((e) => e.path).toList());
                  setState(() {
                    path = res[0].thumbPath;
                  });
                  // bool status = await ImagesPicker.saveImageToAlbum(File(res[0]?.path));
                  // print(status);
                }
              },
            ),
            ElevatedButton(
              child: Text('openCamera'),
              onPressed: () async {
                List<Media>? res = await ImagesPicker.openCamera(
                  pickType: PickType.image,
                  quality: 0.5,
                  // cropOpt: CropOption(
                  //   aspectRatio: CropAspectRatio.wh16x9,
                  // ),
                  // maxTime: 60,
                );
                if (res != null) {
                  print(res[0].path);
                  setState(() {
                    path = res[0].thumbPath;
                  });
                }
              },
            ),
            ElevatedButton(
              onPressed: () async {
                File file = await downloadFile('https://cdn.chavesgu.com/logo.png');
                bool res = await ImagesPicker.saveImageToAlbum(file, albumName: "chaves");
                print(res);
              },
              child: Text('saveNetworkImageToAlbum'),
            ),
            ElevatedButton(
              onPressed: () async {
                File file = await downloadFile('https://cdn.chavesgu.com/SampleVideo.mp4');
                bool res = await ImagesPicker.saveVideoToAlbum(file, albumName: "chaves");
                print(res);
              },
              child: Text('saveNetworkVideoToAlbum'),
            ),
            path != null
                ? Container(
                    height: 200,
                    child: Image.file(
                      File(path!),
                      fit: BoxFit.contain,
                    ),
                  )
                : SizedBox.shrink(),
          ],
        ),
      ),
    );
  }

  Future<File> downloadFile(String url) async {
    Dio simple = Dio();
    String savePath = Directory.systemTemp.path + '/' + url.split('/').last;
    await simple.download(url, savePath, options: Options(responseType: ResponseType.bytes));
    print(savePath);
    File file = new File(savePath);
    return file;
  }
}
